python/pyarrow/_dataset_parquet.pyx

```diff
@@ -317,6 +317,13 @@ cdef class ParquetFileFragment(FileFragment):
             row_groups
         )
 
+    def slice(self, start, end):
+        """
+        Slice is not implemented for Parquet files.
+        """
+        raise Exception("Not Implemented! You cannot slice a Parquet file by byte range.
```

Comment:
`NotImplementedError` is what you are looking for :-)
https://docs.python.org/3/library/exceptions.html#NotImplementedError